cfme/physical/physical_chassis.py

```diff
@@ -0,0 +1,294 @@
+                          "cfme inventory: {})"
+                    raise StatsDoNotMatch(msg.format(inventory, self.name, chass_inventory,
+                                                     cfme_inventory))
+            except KeyError:
+                raise HostStatsNotContains(
+                    "Server inventory information does not contain '{}'".format(inventory))
+            except AttributeError:
+                msg = "Provider does not know how to get '{}'"
+                raise ProviderHasNoProperty(msg.format(inventory))
+
+
+@attr.s
+class PhysicalChassisCollection(BaseCollection):
+    """Collection object for the :py:class:`cfme.physical.physical_chassis.PhysicalChassis`."""
+
+    ENTITY = PhysicalChassis
+
+    def all(self, provider):
+        """
+        Return all `physical_chassis` objects of the given provider.
+
+        Args:
+            provider: the provider that the physical chassis belongs to
+        """
+        physical_chassis_table = self.appliance.db.client['physical_chassis']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_chassis_query = (
+            self.appliance.db.client.session
+                .query(physical_chassis_table.name, physical_chassis_table.ems_ref, ems_table.name)
+                .join(ems_table, physical_chassis_table.ems_id == ems_table.id))
+
+        if self.filters.get('provider'):
+            physical_chassis_query = physical_chassis_query.filter(ems_table.name == provider.name)
+        physical_chassiss = []
+        for name, ems_ref, ems_name in physical_chassis_query.all():
+            physical_chassiss.append(self.instantiate(name=name, ems_ref=ems_ref,
+                                    provider=provider or get_crud_by_name(ems_name)))
+        return physical_chassiss
+
+
+class ComputePhysicalInfrastructureChassisView(BaseLoggedInPage):
+    """Common parts for chassis views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    @property
+    def in_compute_physical_infrastructure_chassis(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ["Compute", "Physical Infrastructure",
+                                                       "Chassis"])
+
+
+class PhysicalChassisEntity(JSBaseEntity):
+    @property
+    def data(self):
+        data_dict = super(PhysicalChassisEntity, self).data
+        if 'quadicon' in data_dict and data_dict['quadicon']:
+            quad_data = document_fromstring(data_dict['quadicon'])
+            data_dict['no_host'] = int(quad_data.xpath(self.QUADRANT.format(pos="a"))[0].text)
+        return data_dict
+
+
+class PhysicalChassisDetailsToolbar(View):
+    """Represents physical chassis toolbar and its controls."""
+    identify = Dropdown(text="Identify")
+
+
+class PhysicalChassisDetailsEntities(View):
+    """Represents Details page Entities."""
+    properties = SummaryTable(title="Properties")
```

Comment:
SummaryTable approach is deprecated in favor of ParametrizedSummaryTable, see here for example https://github.com/ManageIQ/integration_tests/pull/6802. In general, this means you needn't list every piece separately but rather only:

```
summary = ParametrizedSummaryTable
```

and then use it like https://github.com/ManageIQ/integration_tests/blob/4757297e7e3bcc90b747470636b7d958e360857f/cfme/networks/provider/nuage.py#L124-L127

Reply:
👍